Sources/CNIOOpenSSL/helpers.c

```diff
@@ -0,0 +1,82 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2018 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+#include <assert.h>
+#include <openssl/bio.h>
+
+/// This is a pointer to the BIO_METHOD structure for NIO's ByteBufferBIO.
+///
+/// This structure is always initialized at startup, and must be initialized in a
+/// thread-safe manner. That means it should be guarded by some kind of pthread_once
+/// setup behaviour, or a lock. For NIO, we use the initializeOpenSSL dance to do
+/// this construction.
+BIO_METHOD *CNIOOpenSSL_ByteBufferBIOMethod = NULL;
+
+/// This is the type of the ByteBufferBIO created by Swift.
+///
+/// This type is used to create the BIO in Swift code. It can also be used to gate
+/// initialization: if this is non-zero, we have already initialized ByteBufferBIOMethod
+/// and can safely use it.
+int CNIOOpenSSL_ByteBufferBIOType = 0;
+
+static const char *bio_name = "ByteBuffer BIO";
```

Comment:
nit, you're missing one `const` here to make it a true constant: `static const char * const bio_name = "ByteBuffer BIO";`